src/elements/element.rectangle.js

```diff
@@ -78,70 +83,87 @@ module.exports = Element.extend({
 			bottom = vm.y + vm.height / 2;
 			signX = right > left ? 1 : -1;
 			signY = 1;
-			borderSkipped = vm.borderSkipped || 'left';
-		}
-
-		// Canvas doesn't allow us to stroke inside the width so we can
-		// adjust the sizes to fit if we're setting a stroke on the line
-		if (borderWidth) {
-			// borderWidth shold be less than bar width and bar height.
-			var barSize = Math.min(Math.abs(left - right), Math.abs(top - bottom));
-			borderWidth = borderWidth > barSize ? barSize : borderWidth;
-			var halfStroke = borderWidth / 2;
-			// Adjust borderWidth when bar top position is near vm.base(zero).
-			var borderLeft = left + (borderSkipped !== 'left' ? halfStroke * signX : 0);
-			var borderRight = right + (borderSkipped !== 'right' ? -halfStroke * signX : 0);
-			var borderTop = top + (borderSkipped !== 'top' ? halfStroke * signY : 0);
-			var borderBottom = bottom + (borderSkipped !== 'bottom' ? -halfStroke * signY : 0);
-			// not become a vertical line?
-			if (borderLeft !== borderRight) {
-				top = borderTop;
-				bottom = borderBottom;
-			}
-			// not become a horizontal line?
-			if (borderTop !== borderBottom) {
-				left = borderLeft;
-				right = borderRight;
-			}
+			borderSkipped = valueOrDefault(vm.borderSkipped, {left: true}) || {};
 		}
 
-		ctx.beginPath();
 		ctx.fillStyle = vm.backgroundColor;
 		ctx.strokeStyle = vm.borderColor;
-		ctx.lineWidth = borderWidth;
-
-		// Corner points, from bottom-left to bottom-right clockwise
-		// | 1 2 |
-		// | 0 3 |
-		var corners = [
-			[left, bottom],
-			[left, top],
-			[right, top],
-			[right, bottom]
-		];
-
-		// Find first (starting) corner with fallback to 'bottom'
-		var borders = ['bottom', 'left', 'top', 'right'];
-		var startCorner = borders.indexOf(borderSkipped, 0);
-		if (startCorner === -1) {
-			startCorner = 0;
+
+		if (!borderWidth) {
+			ctx.fillRect(left, bottom, right - left, top - bottom);
+			return;
 		}
 
-		function cornerAt(index) {
-			return corners[(startCorner + index) % 4];
+		if (typeof borderSkipped === 'string') {
+			borderSkipped = {};
+			borderSkipped[vm.borderSkipped] = true;
 		}
 
-		// Draw rectangle from 'startCorner'
-		var corner = cornerAt(0);
-		ctx.moveTo(corner[0], corner[1]);
+		maxWidth = Math.abs(left - right) / 2;
+		maxHeight = Math.abs(top - bottom) / 2;
+		if (!helpers.isObject(borderWidth)) {
+			borderWidth = {
+				bottom: borderSkipped.bottom ? 0 : Math.min(borderWidth, maxHeight),
+				left: borderSkipped.left ? 0 : Math.min(borderWidth, maxWidth),
+				top: borderSkipped.top ? 0 : Math.min(borderWidth, maxHeight),
+				right: borderSkipped.right ? 0 : Math.min(borderWidth, maxWidth)
+			};
+		} else {
+			borderWidth = {
+				bottom: Math.min(borderWidth.bottom || 0, maxHeight),
```

Comment:
Don't we need to check `borderSkipped` here as well?

Reply:
I made "the new way" stronger, so you don't have to specify `borderSkipped` if you do specify `borderWidth` as object.

I'm quite not sure if `borderSkipped` should be left as a one border skipper still, because its easier to skip one border that way.

```
borderWidth: 3,
borderSkipper: 'right',
```
vs
```
borderWidth: { left: 3, top: 3, bottom: 3 },
```

Reply:
What if the user specifies:

```
borderWidth: { left: 10, right: 10, top: 5, bottom: 5 },
borderSkipped: { left: true }
```

The way you mentioned would be a more straight forward way of setting it, but it would still make sense that if they set it this way it should have an effect

Reply:
@kurkle not sure if you meant to resolve this thread?

Reply:
No, was the previous one that was resolved. 

I removed the ability to specify multiple borders skipped. But the question still remains, should the one border that's marked to be skipped (by default too), override a `borderWidth` definition for that same border? 

I'd say no, because it defaults to bottom/left.

We could handle a non-default value differently, but that would be inconsistent.

Reply:
The default is `borderWidth: 'bottom'`, so if the `borderWidth` is an object we know it's not the default. I'm not sure I understand the concern or that we need to worry about defaults here

Reply:
`borderWidth: {bottom:1, top:1}`
Vertical chart, should that bottom border be skipped as `borderSkipped: 'bottom'` comes as default?

Reply:
Oh, I see. Ok, that makes sense